packages/payment/src/services/payment-module.ts

```diff
@@ -207,38 +207,33 @@ export default class PaymentModuleService<
     data: CreatePaymentSessionDTO,
     @MedusaContext() sharedContext?: Context
   ): Promise<PaymentSessionDTO> {
-    const created = await this.paymentSessionService_.create(
-      {
+    const { result, errors } = await paymentSessionWorkflow(
+      this.__container__ as any
```

Comment:
I dont see a way of doing that as the context is not shared at any point, it is an outside function and unless binding this and making the workflow aware of this property i can't see a nice and clean way of doing this. You would have to do something at somepoint as we already discussed that for the route as well 👍

Reply:
Yeah, I think you are right. But it would be cool to see if we can find something more intuitive than `this.__container__`. 

Properties that have this type of naming signals "don't touch" in my opinion 😅

Reply:
I can name it differently but it was more to avoid conflicts if a user want to store the container on this.container 🤣

Reply:
At the moment I don't have an anwers how we could skip from passing the container here. maybe sharing the "this" at the sharedContext, so you could get the instance from the context in the workflow step, instead of getting the container

Reply:
Yeah but then we mix responsibilities and the workflow start to know what the service is holding which seems weird

Reply:
but this is literally for the internal workflow. we can't do that outside this scope.

Reply:
Yes true, it is a local module worklow

Reply:
we can also remove the `InjectTransactionManager` from this method so we don't have a long transaction running